core/chain_makers.go

```diff
@@ -104,7 +104,12 @@ func (b *BlockGen) AddTxWithChain(bc *BlockChain, tx *types.Transaction) {
 		b.SetCoinbase(common.Address{})
 	}
 	b.statedb.Prepare(tx.Hash(), len(b.txs))
-	fees := new(big.Int)
+	var fees *big.Int
+	if b.header.Fees != nil {
+		fees = b.header.Fees // wemix block has `Fees` field
+	} else {
+		fees = new(big.Int)
+	}
```

Comment:
```suggestion
	fees := new(big.Int)
	if b.header.Fees != nil {
		fees = b.header.Fees // wemix block has `Fees` field
	}
```
and i wonder `fees = b.header.Fees` is intended code.
because `b.header.Fees` is pointer (*big.Int),
if `fees` value changes, `b.header.Fees` value also changes
In ApplyTransaction fucntion, `fee` value changes by `fees.Add(fees, result.Fee)` this code line.
Therefore if `b.header.fees` value is also not intended to change by changing `fee` value, it looks better to change `fees = b.header.Fees` line to fees = `new(big.Int).Set(b.header.Fees)`

Reply:
Yes, I intended it. `b.header.Fees` must be modified in `ApplyTransaction`.
And the current code looks better to prevent unnecessary `big.Int` allocation.